File: comu/com.energyict.mdc.device.data.rest/src/main/java/com/energyict/mdc/device/data/rest/impl/DeviceGroupInfo.java
/*
 * Copyright (c) 2017 by Honeywell International Inc. All Rights Reserved
 */

package com.energyict.mdc.device.data.rest.impl;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import java.util.List;

@JsonIgnoreProperties(ignoreUnknown = true)
public class DeviceGroupInfo {

    public long id;
    public String mRID;
    public String name;
    public long version;
    public boolean dynamic;
    public String filter;
    public List<Long> devices;
}

File: coko/com.elster.jupiter.yellowfin.extjs/src/main/web/js/yfn/src/model/DeviceGroupInfo.js
/*
 * Copyright (c) 2017 by Honeywell International Inc. All Rights Reserved
 */

/**
 * @class Yfn.model.DeviceGroupInfo
 */
Ext.define('Yfn.model.DeviceGroupInfo', {
    extend: 'Ext.data.Model',
    fields: [
        'name',
        'dynamic',
        {
            name: 'value1',
            type: 'string',
            convert: function(v, record){
                v = record.get('name');
                if(v.indexOf("__##SEARCH_RESULTS##__")!=-1){
                    return Uni.I18n.translate('generatereport.searchResults', 'YFN', 'Search results')
                }
                return Ext.String.htmlEncode(v);
            }
        },
        {
            name: 'value2',
            type: 'string',
            convert: function(v, record){
                return record.get('name');
            }
        }
    ]
});